sdks/python/apache_beam/io/gcp/bigtable_io_test.py

```diff
@@ -0,0 +1,174 @@
+import random
+import string
+import unittest
+import uuid
+
+import apache_beam as beam
+from apache_beam.io.gcp.bigtable_io_write import BigtableWriteConfiguration
+from apache_beam.io.gcp.bigtable_io_write import WriteToBigtable
+from apache_beam.metrics.metric import MetricsFilter
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.runners.runner import PipelineState
+from apache_beam.testing.test_pipeline import TestPipeline
+
+# Protect against environments where bigtable library is not available.
+# pylint: disable=wrong-import-order, wrong-import-position
+try:
+  from google.cloud.bigtable import row, column_family, Client
+except ImportError:
+  Client = None
+
+
+class GenerateDirectRows(beam.DoFn):
+  """ Generates an iterator of DirectRow object to process on beam pipeline.
+
+  """
+  def process(self, row_values):
+    """ Process beam pipeline using an element.
+
+    :type row_value: dict
+    :param row_value: dict: dict values with row_key and row_content having
+    family, column_id and value of row.
+    """
+    direct_row = row.DirectRow(row_key=row_values["row_key"])
+
+    for row_value in row_values["row_content"]:
+      direct_row.set_cell(row_value["column_family_id"],
+                          row_value["column_id"],
+                          row_value["value"],
+                          datetime.datetime.now())
+
+
+@unittest.skipIf(Client is None, 'GCP Bigtable dependencies are not installed')
+class BigtableIOWriteIT(unittest.TestCase):
+  """ Bigtable Write Connector Test
+
+  """
+  DEFAULT_TABLE_PREFIX = "python-test"
+  instance_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  cluster_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  table_id = DEFAULT_TABLE_PREFIX + "-" + str(uuid.uuid4())[:8]
+  number = 500
+  LOCATION_ID = "us-east1-b"
+
+  def setUp(self):
+    try:
+      from google.cloud.bigtable import enums
+      self.STORAGE_TYPE = enums.StorageType.HDD
+    except ImportError:
+      self.STORAGE_TYPE = 2
+
+    self.test_pipeline = TestPipeline(is_integration_test=True)
+    self.runner_name = type(self.test_pipeline.runner).__name__
+    self.project = self.test_pipeline.get_option('project')
+    self.client = Client(project=self.project, admin=True)
+    self._create_instance_table()
+
+  def tearDown(self):
+    instance = self.client.instance(self.instance_id)
+    if instance.exists():
+      instance.delete()
+
+  def test_bigtable_write_python(self):
+    number = self.number
+    config = BigtableWriteConfiguration(self.project, self.instance_id,
+                                        self.table_id)
+    pipeline_args = self.test_pipeline.options_list
+    pipeline_options = PipelineOptions(pipeline_args)
+
+    row_values = self._generate_mutation_data(number)
+
+    with beam.Pipeline(options=pipeline_options) as pipeline:
+      _ = (
+          pipeline
+          | 'Generate Row Values' >> beam.Create(row_values)
```

Comment:
For simplicity, can we combine `'Generate Row Values'` and `'Generate Direct Rows'`?

Reply:
I changed the class `Generate Direct Rows`  to a PTransform.
```python
class GenerateDirectRows(beam.PTransform):
  """ Generates an iterator of DirectRow object to process on beam pipeline.

  """
  def __init__(self, number, **kwargs):
    super(GenerateDirectRows, self).__init__(**kwargs)
    self.number = number
    self.rand = random.choice(string.ascii_letters + string.digits)
    self.column_family_id = 'cf1'

  def _generate(self):
    value = ''.join(self.rand for i in range(100))

    for index in range(self.number):
      key = "beam_key%s" % ('{0:07}'.format(index))
      direct_row = row.DirectRow(row_key=key)
      for column_id in range(10):
        direct_row.set_cell(self.column_family_id,
                            ('field%s' % column_id).encode('utf-8'),
                            value,
                            datetime.datetime.now())
      yield direct_row

  def expand(self, pvalue):
    return (pvalue
            | beam.Create(self._generate()))
```